ykcompile/src/lib.rs

```diff
@@ -503,6 +507,109 @@ impl<TT> TraceCompiler<TT> {
         Ok(())
     }
 
+    fn c_checked_binop(
+        &mut self,
+        dest: &Place,
+        binop: &BinOp,
+        op1: &Operand,
+        op2: &Operand,
+    ) -> Result<(), CompileError> {
+        // Move `op1` into `dest`.
+        match op1 {
+            Operand::Place(p) => self.mov_place_place(dest, &p)?,
+            Operand::Constant(Constant::Int(ci)) => self.mov_place_constint(dest, &ci)?,
+            Operand::Constant(Constant::Bool(b)) => self.mov_place_bool(dest, *b)?,
+            Operand::Constant(c) => return Err(CompileError::Unimplemented(format!("{}", c))),
+        };
+        // Add together `dest` and `op2`.
+        let lloc = self.place_to_location(dest)?;
+        match op2 {
+            Operand::Place(p) => {
+                let rloc = self.place_to_location(&p)?;
+                match binop {
+                    BinOp::Add => self.checked_add_place(lloc, rloc),
+                    _ => todo!(),
+                }
+            }
+            Operand::Constant(Constant::Int(ci)) => match binop {
+                BinOp::Add => self.checked_add_const(lloc, ci),
+                _ => todo!(),
+            },
+            Operand::Constant(Constant::Bool(_b)) => todo!(),
+            Operand::Constant(c) => return Err(CompileError::Unimplemented(format!("{}", c))),
+        };
+        // TODO check for overflow with `jo <label>`
```

Comment:
This is a very important check. For cg_clif many things in libcore broke until I performed overflow checks. Especially in the float parse and display implementation.

Reply:
See d434138